test/src/tests.js

```diff
@@ -320,6 +321,210 @@ describe('Heap Forwarder', function () {
+                Variant: 'variant',
+                Category: 'category',
+                Position: 1,
+                CouponCode: 'coupon',
+                TotalAmount: 999,
+                Attributes: {
+                    prod1AttrKey1: 'value1',
+                    prod1AttrKey2: 'value2',
+                },
+        };
+
+        var product2 = {
+            Name: 'galaxy',
+            Sku: 'galaxySKU',
+            Price: 799,
+            Quantity: 1,
+            Brand: 'brand',
+            Variant: 'variant',
+            Category: 'category',
+            Position: 1,
+            CouponCode: 'coupon',
+            TotalAmount: 799,
+            Attributes: {
+                prod2AttrKey1: 'value1',
+                prod2AttrKey2: 'value2',
+            },
+        };
+
+        var purchaseEvent =  {
+            EventName: 'Test Purchase Event',
+            EventDataType: MessageType.Commerce,
+            EventCategory: EventType.ProductPurchase,
+            ProductAction: {
+                ProductActionType: ProductActionType.Purchase,
+                ProductList: [
+                    product1,
+                    product1,
+                ],
+                TransactionId: 123,
+                Affiliation: 'my-affiliation',
+                TotalAmount: 450,
+                TaxAmount: 40,
+                ShippingAmount: 10,
+                CouponCode: null
+            }
+        };
+
+        var impressionEvent =  {
+            EventName: 'eCommerce - Impression',
+            EventDataType: 16,
+            EventCategory: 22,
+            ProductImpressions: [
+                {
+                    ProductImpressionList: 'Suggested Products List1',
+                    ProductList: [
+                        product1,
+                        product2,
+                    ],
+                },
+                {
+                    ProductImpressionList: 'Suggested Products List2',
+                    ProductList: [
+                        product1,
+                        product2,
+                    ],
+                },
+            ],
+        };
+
+        var promotionEvent = {
+            EventName: 'eCommerce - PromotionClick',
+            EventDataType: 16,
+            CurrencyCode: null,
+            EventCategory: 19,
+            PromotionAction: {
+                PromotionActionType: 2,
+                PromotionList: [
+                    {
+                        Id: 'my_promo_1',
+                        Creative: 'sale_banner_1',
+                        Position: 'top'
+                    },
+                    {
+                        Id: 'my_promo_2',
+                        Creative: 'sale_banner_2',
+                        Position: 'bottom',
+                    },
+                ],
+            },
+        };
+
+        var validateProductProperties = function (properties) {
+            if (properties.product_name === 'iphone') {
```

Comment:
I would avoid using if/else logic inside a test since IMO it introduces unnecessary complexity for what should be a simple matcher.

If you need to reuse these same assertions, I would create two validator methods called `validateiPhoneProperties` and `validateGalaxyProperties`.

As an alternative, you could possibly avoid `validateProductProperties` and simply use:

```
var expectedProperties = {
    prod1AttrKey1: 'value1',
    prod1AttrKey2: 'value2',
    // other attributes.
};
properties.should.deep.equal(expectedProperties);
```

Reply:
I updated this to eliminate some of the conditional by utilizing a single product multiple times and mapped out expected properties. Let me know if this needs further tweaks.